### PR TITLE
Scripts/Quest: Captain Tyralius improvements

### PR DESCRIPTION
**Pull request by Wyreth**
**Changes proposed:**

- Spawn Captain Tyralius and related trigger NPC together with the prison.
- Spell visuals, timers, shorter respawn time due to being a quest-related object.

**Target branch(es):**

- [x] 3.3.5
- [x] master

**Issues addressed:** Updates #12352.

**Tests performed:** works fine.

### CODE REVIEW

File: src/server/scripts/Outland/zone_netherstorm.cpp [hunk 1]
```diff
@@ -680,15 +685,26 @@ class go_captain_tyralius_prison : public GameObjectScript
...
-                me->UseDoorOrButton();
+                me->SetRespawnTime(60);
+                me->SetLootState(GO_JUST_DEACTIVATED);
+
                 if (Creature* tyralius = me->FindNearestCreature(NPC_CAPTAIN_TYRALIUS, 1.0f))
                 {
                     player->KilledMonsterCredit(NPC_CAPTAIN_TYRALIUS);
-                    tyralius->AI()->Talk(SAY_FREE);
-                    tyralius->DespawnOrUnsummon(8000);
+                    tyralius->AI()->DoAction(0);
```
Comment (Keader) [inline src/server/scripts/Outland/zone_netherstorm.cpp]: magic number

File: src/server/scripts/Outland/zone_netherstorm.cpp [hunk 2]
```diff
@@ -699,6 +715,50 @@ class go_captain_tyralius_prison : public GameObjectScript
...
+        return new npc_captain_tyraliusAI(creature);
+    }
+
+    struct npc_captain_tyraliusAI : public ScriptedAI
+    {
+        npc_captain_tyraliusAI(Creature* creature) : ScriptedAI(creature) { }
+
+        void DoAction(int32 /*action*/) override
+        {
+            _events.ScheduleEvent(EVENT_TALK, 0);
```
Comment (Keader) [inline src/server/scripts/Outland/zone_netherstorm.cpp]: it's correct? just 0? looks weird to me

File: src/server/scripts/Outland/zone_netherstorm.cpp [hunk 3]
```diff
@@ -699,6 +715,50 @@ class go_captain_tyralius_prison : public GameObjectScript
...
+
+        void UpdateAI(uint32 diff) override
+        {
+            _events.Update(diff);
+
+            if (uint32 eventId = _events.ExecuteEvent())
+            {
+                switch (eventId)
+                {
+                case EVENT_TALK:
```
Comment (Keader) [inline src/server/scripts/Outland/zone_netherstorm.cpp]: identation

File: src/server/scripts/Outland/zone_netherstorm.cpp [hunk 4]
```diff
@@ -699,6 +715,50 @@ class go_captain_tyralius_prison : public GameObjectScript
...
+        npc_captain_tyraliusAI(Creature* creature) : ScriptedAI(creature) { }
+
+        void DoAction(int32 /*action*/) override
+        {
+            _events.ScheduleEvent(EVENT_TALK, 0);
+        }
+
+        void UpdateAI(uint32 diff) override
+        {
+            _events.Update(diff);
```
Comment (Keader) [inline src/server/scripts/Outland/zone_netherstorm.cpp]: Missing a if (!UpdateVictim()) check here
And this creature should attack something?

File: sql/updates/world/3.3.5/2017_09_25_99_world_captain_tyralius.sql [hunk 5]
```diff
@@ -0,0 +1,4 @@
+--
+UPDATE `creature_template` SET `unit_flags`=512, ScriptName="npc_captain_tyralius" WHERE `entry`=20787;
```
Comment (Jildor) [inline sql/updates/world/3.3.5/2017_09_25_99_world_captain_tyralius.sql]:
```diff
-UPDATE `creature_template` SET `unit_flags`=512, ScriptName="npc_captain_tyralius" WHERE `entry`=20787;
+UPDATE `creature_template` SET `unit_flags`=512,`ScriptName`='npc_captain_tyralius' WHERE `entry`=20787;

File: src/server/scripts/Outland/zone_netherstorm.cpp (hunk 4 above)
Comment (Wyreth) [reply src/server/scripts/Outland/zone_netherstorm.cpp]: Nope, it's an always-friendly creature. Never attacks anything.

File: src/server/scripts/Outland/zone_netherstorm.cpp (hunk 2 above)
Comment (Wyreth) [reply src/server/scripts/Outland/zone_netherstorm.cpp]: Makes sense, thanks :D done.

File: sql/updates/world/3.3.5/2017_09_25_99_world_captain_tyralius.sql (hunk 5 above)
Comment (Wyreth) [reply sql/updates/world/3.3.5/2017_09_25_99_world_captain_tyralius.sql]: Fixed, thanks :)

File: src/server/scripts/Outland/zone_netherstorm.cpp [hunk 6]
```diff
@@ -668,7 +668,12 @@ class npc_maxx_a_million_escort : public CreatureScript
 enum CaptainTyralius
 {
     NPC_CAPTAIN_TYRALIUS    = 20787,
+    NPC_ETHEREUM_PRISONER   = 20825,
+    SPELL_GREEN_BANISH      = 32567,
```
Comment (Keader) [inline src/server/scripts/Outland/zone_netherstorm.cpp]: not used?

Comment (Wyreth) [reply src/server/scripts/Outland/zone_netherstorm.cpp]: Right, it's from a previous fix attempt. Forgot to remove that, thanks!

**Pull request merged**